SU-HSDO/suhumsci
# STN-379: Sprint 17 Feedback

# READY FOR REVIEW

## Summary
This PR fixes all of the issues found in the Sprint 17 QA and cleans up some unused styles.

_QA Fixes:_
- form iframes fill the entire width of a container and have a larger height
- video iframes fill the entire width of a container and maintian a 16:9 aspect ratio
- list items flow along with floated content
- bullet points and numbers no longer overlay floated images
- wysiwyg tables and table views have margin bottom
- Traditional secondary button has a more noticable hover state
- When an h2-h6 is the only element in a row well all margin is removed to "vertically center" the heading in the well
- text areas in row wells now will the width of the container even when the content doesn't
- FA email icons are hidden by default and can be added with a utility class `hb-mailto-link`
- vertical card patterns now display the author and year fields
- subcontent fields bug where a subcontent field didn't show when only one field was mapped has been fixed

_Code Cleanup:_
- styles were originally added to remove the external icon in various scss partials but are no longer necessary since we are now hiding the icon by default

## Need Review By (Date)
asap

## Urgency
medium

## Steps to Test
1. Check that all tests are passing
1. - [x] Check that there are no external icons showing up in places that are not using the `hb-external-link` utility class.
1. - [x] Modify the vertical card on the [Publications](http://economics.suhumsci.loc/research/publications) page to map the author and publication year to the author and year fields (note: there are a few other changes required for this one, I'll export the view config and send it your way).
    <img width="267" alt="Screen Shot 2020-06-09 at 9 40 08 AM" src="https://user-images.githubusercontent.com/2486846/84154534-3bb61a00-aa35-11ea-9d1f-3fd4c1b5ea52.png">
    Multiple authors should display in a comma separated list and the year should display below the author field.
1. - [x] In the ☝️ same view map a single field to the subcontent field and confirm that a single view still displays.
1. - [x] Checkout the [QA Form](http://economics.suhumsci.loc/qa/qa-form) page and confirm that the iframe fills the entire width of the content area and is 70vh high.
1. - [x] Add a video to the page and confirm that the iframe fills the entire width of the content area and maintains a 16:9 aspect ratio.
1. - [x] On the [Image Styles](http://economics.suhumsci.loc/qa/qa-image-styles) page add unordered lists and confirm that:
    - When an image is floated to the right the list items flow under the floated images
        <img width="995" alt="Screen Shot 2020-06-09 at 9 44 57 AM" src="https://user-images.githubusercontent.com/2486846/84155107-ef1f0e80-aa35-11ea-8122-9d312dc78a74.png">
        <img width="985" alt="Screen Shot 2020-06-09 at 9 45 03 AM" src="https://user-images.githubusercontent.com/2486846/84155117-f219ff00-aa35-11ea-888c-61c64ce5c5b2.png">
    - When an image is floated to the left the bullet point/numbers do not lay on top of the image in both the wysiwyg editor and on the front end.
1.  - [x] On the [Faculty](http://economics.suhumsci.loc/people/faculty) page the FA email icon in structured cards should not be displayed unless you add the class `hs-mailto-link` to the email fied.
1. - [x] Confirm that email icons do not show up when using a mailto link in a hero or carousel button.
1.  - [x] Create a basic page and add the following and confirm there is correct margin bottom to each of the table elements:
    - A row with table in a text area
    - A row with 2 tables table in a text area
    - A row with the courses table view (to make it easier I modified it to only show 5 table rows)
    - Another row with the courses table view
1. - [x] Create a basic page and add a series of rows with limited and full width wells. Each row should contain a single text area with a single heading tag (h2-h6 will work). When a single heading is in the row the margin on the heading is removed to appear vertically centered.
    <img width="1238" alt="Screen Shot 2020-06-09 at 9 52 39 AM" src="https://user-images.githubusercontent.com/2486846/84155985-f98dd800-aa36-11ea-837a-d64c072dcff2.png">
    <img width="1277" alt="Screen Shot 2020-06-09 at 9 54 28 AM" src="https://user-images.githubusercontent.com/2486846/84156217-3d80dd00-aa37-11ea-86db-eed89a8a9657.png">
1. - [x] In the ☝️ same page adjust the text alignment for a shorter heading and confirm that the text will center and right align.
    <img width="1261" alt="Screen Shot 2020-06-09 at 9 53 56 AM" src="https://user-images.githubusercontent.com/2486846/84156145-26da8600-aa37-11ea-8bb8-864cd24dd6e6.png">


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
